source/slang/slang-check-conversion.cpp

```diff
@@ -92,13 +101,24 @@ namespace Slang
         if(isEffectivelyScalarForInitializerLists(fromExpr->type))
             return false;
 
-        // Once the above cases are handled, the main thing
-        // we want to check for is whether a direct initialization
-        // is possible (a type conversion exists).
-        //
-        return canCoerce(toType, fromExpr->type, fromExpr);
+        // If 2 types are "broadly equal" we know the types can be coerced directly
+        // since otherwise trying to solve `float3 tmp[2] = {float3(val), float3(val)};` will break down into searching for `float`
+        // (which we don't have in our init-list argument list, this will error).
+        // 
+        // This logic should be completly cleaned up at some point by only using constructors for `{...}` and attempting to use 
+        // constructors before fallback coerce logic is tried.
+        if (toType->equals(fromExpr->type))
+            return true;
+
+        // `vector<T,N>` and `vector<T,U>` may not be equal types, but, we have no choice in this senario but to assume coerce logic
+        // can resolve such situations, otherwise we need to fail
+        if (isSameSimpleExprType(toType, fromExpr->type))
```

Comment:
how can this not implied by line 110?

Reply:
if we have 2 aggregate types we return true